core/commonMain/src/implementations/immutableList/AbstractPersistentList.kt

```diff
@@ -22,7 +22,7 @@ abstract class AbstractPersistentList<E> : PersistentList<E>, AbstractList<E>()
     }
 
     override fun removeAll(elements: Collection<E>): PersistentList<E> {
-        val hashSet = elements.toHashSet()
+        val hashSet = elements.convertToSetForSetOperationWith(this)
```

Comment:
I'd ditch this optimization entirely for the sake of a more predictable behavior.

Reply:
done